builtin/credential/approle/path_login.go

```diff
@@ -38,6 +38,9 @@ func (b *backend) pathLoginUpdate(req *logical.Request, data *framework.FieldDat
 		return logical.ErrorResponse(fmt.Sprintf("failed to validate SecretID: %s", err)), nil
 	}
 
+	// Always include the role name, for later filtering
+	metadata["role_name"] = roleName
```

Comment:
@vishalnayak one problem with that approach is that existing roles/secret IDs won't have that metadata value persisted unless they're rewritten, whereas this will always work.